Framework/LiveData/src/ISIS/ISISHistoDataListener.cpp

```diff
@@ -124,6 +124,18 @@ bool ISISHistoDataListener::connect(const Poco::Net::SocketAddress &address) {
 
   loadTimeRegimes();
 
+  // Create dummy workspace to store instrument data
+  bufferWorkspace = WorkspaceFactory::Instance().create("Workspace2D", 1, 1, 1);
+
+  bufferWorkspace->getAxis(0)->unit() =
+      Kernel::UnitFactory::Instance().create("TOF");
+  bufferWorkspace->setYUnit("Counts");
+  bufferWorkspace->updateSpectraUsing(
```

Comment:
I think this does not work as expected anymore: Previously you used `WorkspaceFactory` to create from parent with the correct number of histograms. Now it is set to `1` above. Otherwise, I think `updateSpectraUsing` is not doing what you expect.

Reply:
It's easier to move updateSpectraUsing(...) to the extractData function because it seems the time regime can be updated during the streaming.